pytorch_lightning/trainer/data_loading.py

```diff
@@ -298,13 +328,13 @@ def reset_train_dataloader(self, model: Optional["pl.LightningModule"] = None) -
                     "You requested to overfit but enabled training dataloader shuffling."
                     " We are turning off the training dataloader shuffling for you."
                 )
-                self.train_dataloader = self.replace_sampler(
+                self.train_dataloader = self._prepare_dataloader(
```

Comment:
what if `self.train_dataloader` is not strict Dataloader, but a list of Dataloader? see the following [line351](https://github.com/PyTorchLightning/pytorch-lightning/pull/9721/files#diff-15fd4659acd10d0716002ba2dbfd67aca83dc101d44531c2f234f65140e1e39dR351). 
This is unrelated to this PR, but `reset_train_dataloader` has a lot of operations, the orders of them matter. We need to revisit them to see what is the reasonable order.

Reply:
Good point. Replace to apply_to_collection.